tree.go

```diff
@@ -820,24 +725,33 @@ func (n *InternalNode) Serialize() ([]byte, error) {
 			}
 		}
 	}
-	children := make([]byte, 0, (len(commitments)+nhashed)*32)
 
-	bytecomms := banderwagon.ElementsToBytes(commitments)
+	ret := make([]byte, nodeTypeSize+bitlistSize+(len(commitments)+nhashed)*SerializedPointSize)
```

Comment:
OK, some optimizations happening in this method which makes things lighter for the CPU and produce less allocation & GC pressure.

Before in L823 (deleted), we created the `children` slice and later did some "appending" to produce the final serialized result.

Now we do something smarter. I create the `ret` byte slice to be returned here, and everything will be calculated directly in this slice.